src/solver/optim-model-filler/include/antares/solver/optim-model-filler/LinearExpression.h

```diff
@@ -97,21 +97,21 @@ using FullKeyMap = std::unordered_map<FullKey, double, FullKeyHash>;
  * ```
  */
 template<typename MapType, typename UnaryOp = std::identity>
-MapType add_maps(const MapType& left, const MapType& right, UnaryOp op = std::identity{})
+void add_maps(MapType& left, const MapType&& right, UnaryOp op = std::identity{})
 {
-    auto result(left);
-    for (auto [key, value]: right)
+    for (auto& [key, value]: right)
     {
-        if (result.contains(key))
+        if (left.find(key) != left.end())
```

Comment:
not efficient, you should find the element outside the if then compare it to .end()